servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Single.java

```diff
@@ -1057,6 +1057,96 @@ public final Completable flatMapCompletable(Function<? super T, ? extends Comple
+     * the downstream {@link Subscriber} subscribed will not be queued.
+     * <p>
+     * In sequential programming this is similar to the following:
+     * <pre>{@code
+     *     T result = resultOfThisSingle();
+     *     List<T> multiResults = ...; // simulating multiple Subscribers
+     *     for (int i = 0; i < expectedSubscribers; ++i) {
+     *         multiResults.add(result);
+     *     }
+     *     return multiResults;
+     * }</pre>
+     * @param minSubscribers The upstream subscribe operation will not happen until after this many {@link Subscriber}
+     * subscribe to the return value.
+     * @return a {@link Single} that subscribes a single time upstream but allows for multiple downstream
+     * {@link Subscriber}s. Signals from upstream will be multicast to each downstream {@link Subscriber}.
+     * @see <a href="http://reactivex.io/documentation/operators/publish.html">ReactiveX multicast operator</a>
+     * @see #multicast(int, boolean)
+     */
+    public final Single<T> multicast(int minSubscribers) {
+        return multicast(minSubscribers, true);
+    }
+
+    /**
+     * Create a {@link Single} that subscribes a single time upstream but allows for multiple downstream
+     * {@link Subscriber}s. Signals from upstream will be multicast to each downstream {@link Subscriber}.
+     * <p>
+     * Downstream {@link Subscriber}s may subscribe after the upstream subscribe, but signals that were delivered before
+     * the downstream {@link Subscriber} subscribed will not be queued.
+     * <p>
+     * In sequential programming this is similar to the following:
+     * <pre>{@code
+     *     T result = resultOfThisSingle();
+     *     List<T> multiResults = ...; // simulating multiple Subscribers
+     *     for (int i = 0; i < expectedSubscribers; ++i) {
+     *         multiResults.add(result);
+     *     }
+     *     return multiResults;
+     * }</pre>
+     * @param minSubscribers The upstream subscribe operation will not happen until after this many {@link Subscriber}
+     * subscribe to the return value.
+     * @param cancelUpstream {@code true} if upstream should be {@link Cancellable#cancel() cancelled} when all
+     * downstream {@link Subscriber}s cancel. {@code false} means that cancel will not be propagated upstream even if
+     * all downstream {@link Subscriber}s cancel, and the upstream Subscription will stay valid until termination.
+     * @return a {@link Single} that subscribes a single time upstream but allows for multiple downstream
+     * {@link Subscriber}s. Signals from upstream will be multicast to each downstream {@link Subscriber}.
+     * @see <a href="http://reactivex.io/documentation/operators/publish.html">ReactiveX multicast operator</a>
+     * @see #multicast(int, boolean, BiFunction)
+     */
+    public final Single<T> multicast(int minSubscribers, boolean cancelUpstream) {
+        return multicast(minSubscribers, cancelUpstream, (v, t) -> Completable.never());
```

Comment:
`(v, t) -> Completable.never()` -> the default here is different than what is used for `Publisher` (which uses `Completable.completed()` which clears state to allow for immediate re-subscribes by default). The difference is that we don't have to worry about data being dropped bcz for Single the terminal event includes all the data, and re-subscribes can get access to the complete state. Open to adjusting for consistency though.

Reply:
this is now not an issue after renaming to `cache` as the difference in behavior is expected.